validator/src/main.rs

```diff
@@ -392,6 +392,20 @@ fn hash_validator(hash: String) -> Result<(), String> {
         .map_err(|e| format!("{:?}", e))
 }
 
+/// Returns the default shred storage size (include both data and coding
+/// shreds) based on the validator config.
+fn default_fifo_shred_storage_size(vc: &ValidatorConfig) -> u64 {
+    // This is an estimation that is slightly larger than average so that we
+    // can make sure FIFO rarely evits data/coding shreds that is younger
+    // than the configured max_ledger_shreds while avoiding allocating too
+    // much disk space for data/coding shreds.
+    const AVG_SHRED_SIZE_BYTE: u64 = 3200;
+    match vc.max_ledger_shreds {
+        Some(max_ledger_shreds) => (max_ledger_shreds / 2) * AVG_SHRED_SIZE_BYTE,
```

Comment:
This is a reduction in how many shreds we'll store right? If max_ledger_shreds == 200,000,000 (the default), then:
```
( 200,000,000 / 2 ) * 3200 bytes = 320,000,000,000 bytes = 320 GB
```
That is 320 GB total, so 160 GB per column family. Or, essentially, 100 million data and 100 million coding whereas the current configuration is setup to take 200 million data shreds

Reply:
The way I am thinking this is that: given the default 200,000,000 shreds require the suggested 500GB.  We previously observed that data + coding shreds occupy ~50% of the ledger data (no rpc case).  This gives us ~250GB.

This PR increases to 320GB to give an extra storage buffer to FIFO to prevent the case where FIFO cleans up data earlier than our ledger cleanup service.

Reply:
The updated code snippet:
```rust
    const RESERVED_BYTES_PER_SHRED: u64 = 1600;
    match vc.max_ledger_shreds {
        Some(max_ledger_shreds) => max_ledger_shreds * RESERVED_BYTES_PER_SHRED,
```
With defaults limit-ledger-size value (200 million):
```
200,000,000 shreds * 1600 bytes / shred = 320,000,000,00 = 320 GB
```
That output of above function is consumed here:
```rust
                "fifo" => match matches.value_of("rocksdb_fifo_shred_storage_size") {
                    None => ShredStorageType::rocks_fifo(default_fifo_shred_storage_size(
                        &validator_config,
                    )),
```
`ShredStorageType::rocks_fifo()` implementation:
```rust
impl ShredStorageType {
    /// Returns ShredStorageType::RocksFifo where the specified
    /// `shred_storage_size` is equally allocated to shred_data_cf_size
    /// and shred_code_cf_size.
    pub fn rocks_fifo(shred_storage_size: u64) -> ShredStorageType {
        ShredStorageType::RocksFifo(BlockstoreRocksFifoOptions::new(shred_storage_size))
    }
```
`BlockstoreRocksFifoOptions::new()` implementation:
```rust
impl BlockstoreRocksFifoOptions {
    fn new(shred_storage_size: u64) -> Self {
        Self {
            shred_data_cf_size: shred_storage_size / 2,
            shred_code_cf_size: shred_storage_size / 2,
        }
    }
}
```

That 320 GB is getting divided up at the bottom here, so 160 GB per column right?

Reply:
> That 320 GB is getting divided up at the bottom here, so 160 GB per column right?

Yes, it is 160GB per column.

Reply:
Oh, oh, oh!  I was indeed wrong.  It is ~50% if we consider RPC calls, and nearly 95% if we don't consider RPC calls.

So it would be 320GB per column if we want to add some extra buffer.  Maybe 300GB each is already enough.  I will update the PR and the description of the argument to include this information.